app/api/services/suppliers.py

```diff
@@ -43,6 +43,23 @@ def get_supplier_by_code(self, code):
             .one_or_none()
         )
 
+    def get_supplier_by_abn(self, abn):
+        return (
+            db
+            .session
+            .query(
+                Supplier
+            )
+            .options(
+                joinedload(Supplier.domains)
+                .joinedload(SupplierDomain.domain)
+            )
+            .filter(func.replace(Supplier.abn, ' ', '') == abn)
+            .filter(Supplier.status != 'deleted')
+            .limit(1)
```

Comment:
is this really needed?

Reply:
I believe we have had cases before where we've had duplicate ABNs - there is currently no restriction on this so it's technically possible we have dupes.